File: solutions/2026-01-31/solution.py
# Add One Row to Tree
# Difficulty: Medium
# URL: https://leetcode.com/problems/add-one-row-to-tree/

class Solution:
    def addOneRow(self, root: TreeNode, val: int, depth: int) -> TreeNode:
        if depth == 1:
            return TreeNode(val, root, None)
        self.dfs(root, 1, depth, val)
        return root

    def dfs(self, node, cur_depth, target_depth, val):
        if not node:
            return
        if cur_depth == target_depth - 1:
            node.left = TreeNode(val, node.left, None)
            node.right = TreeNode(val, None, node.right)
        self.dfs(node.left, cur_depth + 1, target_depth, val)
        self.dfs(node.right, cur_depth + 1, target_depth, val)

File: solutions/2026-01-31/solution.java
// Add One Row to Tree
// Difficulty: Medium
// URL: https://leetcode.com/problems/add-one-row-to-tree/

class Solution {
    public TreeNode addOneRow(TreeNode root, int val, int depth) {
        if (depth == 1) {
            return new TreeNode(val, root, null);
        }
        dfs(root, 1, depth);
        return root;
    }

    void dfs(TreeNode node, int curDepth, int targetDepth) {
        if (node == null) {
            return;
        }
        if (curDepth == targetDepth - 1) {
            node.left = new TreeNode(val, node.left, null);
            node.right = new TreeNode(val, null, node.right);
        }
        dfs(node.left, curDepth + 1, targetDepth);
        dfs(node.right, curDepth + 1, targetDepth);
    }
}